packages/sdk/src/services/editLp/editLp.ts

```diff
@@ -0,0 +1,194 @@
+import { BigNumber, ContractReceipt } from 'ethers';
+import {
+  estimateGas,
+  executeTransaction,
+  simulateTx,
+} from '../executeTransaction';
+import {
+  getNativeGasToken,
+  convertGasUnitsToNativeTokenUnits,
+} from '@voltz-protocol/sdk-v1-stateless';
+import { getTokenDetails, scale, descale } from '@voltz-protocol/commons-v2';
+import { notionalToLiquidityBN } from '../../utils/helpers';
+import { defaultAbiCoder } from 'ethers/lib/utils';
+import { CompleteEditLpDetails, LpArgs, LpPeripheryParameters } from './types';
+import { getEditLpPeripheryParams } from './getEditLpPeripheryParams';
+import { InfoPostLp } from '../lp';
+import { encodeLp } from '../lp/encode';
+
+async function createLpParams({
+  poolId,
+  positionId,
+  signer,
+  notionalAmount,
+  marginAmount,
+  fixedRateLower,
+  fixedRateUpper,
+}: LpArgs): Promise<CompleteEditLpDetails> {
+  const lpInfo = await getEditLpPeripheryParams(poolId, positionId);
+
+  const tokenDecimals = getTokenDetails(lpInfo.quoteTokenAddress).tokenDecimals;
+  const liquidityAmount = notionalToLiquidityBN(
+    scale(tokenDecimals)(notionalAmount),
+    tokenDecimals,
+    fixedRateLower,
+  );
+
+  const params: CompleteEditLpDetails = {
+    ...lpInfo,
+    owner: signer,
+    liquidityAmount: liquidityAmount,
+    marginAmount: scale(tokenDecimals)(marginAmount),
+    fixedRateLower,
+    fixedRateUpper,
+  };
+
+  return params;
+}
+
+export async function lp({
+  poolId,
+  positionId,
+  signer,
+  notionalAmount,
+  marginAmount,
+  fixedRateLower,
+  fixedRateUpper,
+}: LpArgs): Promise<ContractReceipt> {
+  // fetch: send request to api
+
+  const params = await createLpParams({
+    poolId,
+    positionId,
+    signer,
+    notionalAmount,
+    marginAmount,
+    fixedRateLower,
+    fixedRateUpper,
+  });
+
+  const { data, value, chainId } = await getLpTxData(params);
+  const result = await executeTransaction(signer, data, value, chainId);
+  return result;
+}
+
+export async function getInfoPostLp({
```

Comment:
worth having one getInfoPostLp and one for getInfoPostLpEdit? one requires the poolId, the other requires the positionId to align with v1 unless you think there's a good reason to go with this pattern?